### PR TITLE
feat(terraform): New classes for the TF module model

### PR DESCRIPTION
**Pull request by ChanochShayner**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
New classes for the TF module model.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/utils.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    def __eq__(self, other) -> bool:
+        return self.path == other.path and self.name == other.name and self.foreach_idx == other.foreach_idx
+
+    def __repr__(self) -> str:
+        return f'path:{self.path}, name:{self.name}, foreach_idx:{self.foreach_idx}'
+
+
+class TFSourceModules:
+    __slots__ = "tf_source_modules"
```
Comment (gruebel) [inline checkov/terraform/utils.py:22]:
```suggestion
    __slots__ = ("tf_source_modules",)
```
I think slots needs to be an iterable

File: checkov/terraform/utils.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+class TFModule:
+    __slots__ = ("path", "name", "foreach_idx")
+
+    def __init__(self, path: str, name: str, foreach_idx: Optional[int | str] = None) -> None:
+        self.path = path
+        self.name = name
+        self.foreach_idx = foreach_idx
+
+    def __eq__(self, other) -> bool:
```
Comment (gruebel) [inline checkov/terraform/utils.py:14]:
```suggestion
    def __eq__(self, other: Any) -> bool:
        if not isinstance(other, TFModule):
            return False
```
same for the other two 🙂 better to be safe than sorry 😄

**Review by achiar99 — APPROVED** [review]

File: checkov/terraform/utils.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+from typing import Optional, Any
+
+
+class TFModule:
```
Comment (bo156) [inline checkov/terraform/utils.py:6]: `utils.py` is not a great name.
What about creating a new directory (modules/module_objects) with a file for each class?

**Review by bo156 — CHANGES_REQUESTED** [review]

File: checkov/terraform/utils.py (hunk 3 above)
Comment (ChanochShayner) [reply checkov/terraform/utils.py]: Moved to `modules/module_objects.py` i think that file per class is too much here.

**Review by bo156 — APPROVED** [review]

**Pull request merged**